### PR TITLE
Jwt secret from env

### PR DESCRIPTION
**Pull request by invaliduser**

### CODE REVIEW

**Review by cliffcaseyyet — COMMENTED** [review]
Overall good, though I suspect you are missing a step which is I believe new config needs to be added. The reason it did not flag you is that instrumentation is off at runtime and in unit tests (where instrumentation is on) nothing adds the parameter.

File: src/main/lrsql/spec/config.clj [hunk 1]
```diff
@@ -228,6 +228,7 @@
                     ::jwt-no-val-role
                     ::jwt-no-val-role-key
                     ::jwt-no-val-logout-url
+                    ::jwt-common-secret
```
Comment (cliffcaseyyet) [inline src/main/lrsql/spec/config.clj:231]: but it doesn't exist as a def?

Comment (milt) [reply src/main/lrsql/spec/config.clj:231]: Yeah you need to add this to the config spec, you'll see all this other stuff in there

File: doc/env_vars.md [hunk 2]
```diff
@@ -149,6 +149,7 @@ _NOTE:_ `LRSQL_STMT_RETRY_LIMIT` and `LRSQL_STMT_RETRY_BUDGET` are used to mitig
 | `LRSQL_JWT_NO_VAL_ROLE_KEY`       | `jwtNoValRoleKey`        | (**DANGEROUS!** See `LRSQL_JWT_NO_VAL`) This variable configures which claim key to look in for the role when token validation is turned off.    | Not Set |
 | `LRSQL_JWT_NO_VAL_ROLE`           | `jwtNoValRole`           | (**DANGEROUS!** See `LRSQL_JWT_NO_VAL`) This variable configures what role must be present in the key above when token validation is turned off.    | Not Set |
 | `LRSQL_JWT_NO_VAL_LOGOUT_URL`     | `jwtNoValLogoutUrl`      | (**DANGEROUS!** See `LRSQL_JWT_NO_VAL`) This variable specifies a logout URL that the client will redirect to on user logout when token validation is turned off. | Not Set |
+| `LRSQL_JWT_COMMON_SECRET    `     | `jwtCommonSecret`        |  This variable serves as a secret key with which JWT's are signed. If this is unspecified, sql-lrs generates its own key, but when running multiple instances of sql-lrs behind load balancer, they need to accept each others' JWTs, meaning they need to be working from a common key.  This is a secret and should be secured as such.  It should be between 32 and 64 characters.| Not Set |
```
Comment (milt) [inline doc/env_vars.md:152]: this is hilariously minor but there is a superfluous possessive apostrophe on the first "JWT's"

File: dev-resources/load-balanced-docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      - "8080:8080"
+    depends_on:
+      - db
+    environment:
+      LRSQL_API_KEY_DEFAULT: my_key
+      LRSQL_API_SECRET_DEFAULT: my_secret
+      LRSQL_ADMIN_USER_DEFAULT: my_username
+      LRSQL_ADMIN_PASS_DEFAULT: my_password
+      LRSQL_ALLOW_ALL_ORIGINS: true
+      LRSQL_DB_HOST: db
```
Comment (cliffcaseyyet) [inline dev-resources/load-balanced-docker-compose.yml:36]: couldn't help but notice this config will fail. `Error: missing sandwich`

File: dev-resources/load-balanced-docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      LRSQL_DB_NAME: lrsql_db
+      LRSQL_DB_USER: lrsql_user
+      LRSQL_DB_PASSWORD: lrsql_password
+      # If Postgres is too slow to start, increase this
+      LRSQL_POOL_INITIALIZATION_FAIL_TIMEOUT: 10000
```
Comment (milt) [inline dev-resources/load-balanced-docker-compose.yml:62]: I don't see `LRSQL_JWT_COMMON_SECRET` used in these?

File: doc/env_vars.md (hunk 2 above)
Comment (invaliduser) [reply doc/env_vars.md:152]: beware the mutant, the alien, the heretic

**Review by cliffcaseyyet — COMMENTED** [review]
Spec and doc good. Docker-compose is missing common key so will inherently reproduce the bug rather than be a cluster.

File: dev-resources/load-balanced-docker-compose.yml (hunk 4 above)
Comment (invaliduser) [reply dev-resources/load-balanced-docker-compose.yml:62]: good point, I'd put it in `lrsql.json` for testing, but would be kind to users to put something in.

Comment (milt) [reply dev-resources/load-balanced-docker-compose.yml:62]: I don't see a `lrsql.json` here tho.. definitely give it a working config right in the dockerfile, I wanna try it out after all

Comment (invaliduser) [reply dev-resources/load-balanced-docker-compose.yml]: added JRSQL_JWT_COMMON_SECRET=sandwich

File: dev-resources/load-balanced-docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
+# Runs SQL LRS with Postgres - Provided for demonstration purposes only!
```
[the remark this answers is not in the document]
Comment (invaliduser) [reply dev-resources/load-balanced-docker-compose.yml]: description added (see load_balanced/docker-compose.yml)

File: dev-resources/load-balanced-docker-compose.yml (hunk 3 above)
Comment (invaliduser) [reply dev-resources/load-balanced-docker-compose.yml]: added sandwich

File: dev-resources/load_balanced/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,80 @@
+version: "3.9"
+# README:
+# Runs *two* SQL LRS containers next to each other behind a load balancer(nginx).
+
+# To run: 
+# 1.  Replace the main `lrsql/docker-compose.yml` file with this one
```
Comment (cliffcaseyyet) [inline dev-resources/load_balanced/docker-compose.yml:6]: this instruction is not really relevant. just run docker compose in this dir

**Review by cliffcaseyyet — APPROVED** [review]

**Review by milt — APPROVED** [review]

**Pull request merged**